pkg/health/server.go

```diff
@@ -33,16 +34,56 @@ type Server interface {
 }
 
 type server struct {
-	ready *atomic.Bool
-	log   logger.Logger
+	ready  *atomic.Bool
+	router http.Handler
+	log    logger.Logger
+}
+
+type RouterOptions func(log logger.Logger) (string, http.Handler)
+
+func NewRouterOptions(path string, handler http.Handler) RouterOptions {
+	return func(log logger.Logger) (string, http.Handler) {
+		return path, handler
+	}
+}
+
+func NewJSONDataRouterOptions[T any](path string, getter func() (T, error)) RouterOptions {
+	return func(log logger.Logger) (string, http.Handler) {
+		return path, http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			data, err := getter()
+			if err != nil {
+				writer.WriteHeader(http.StatusInternalServerError)
+				writer.Write([]byte(err.Error()))
+				return
+			}
+			writer.Header().Set("Content-Type", "application/json")
+			writer.WriteHeader(http.StatusOK)
```

Comment:
Remove this, which is implied when the first byte is written. Otherwise, in case of errors, the `StatusInternalError` will never be sent


```suggestion
```